src/main/java/uk/gov/justice/hmpps/casenotes/config/TokenVerifyingAuthManager.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.hmpps.casenotes.config
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpHeaders
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.core.Authentication
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken
+import org.springframework.security.oauth2.server.resource.InvalidBearerTokenException
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+
+@Suppress("SpringJavaInjectionPointsAutowiringInspection")
+@Component
+class TokenVerifyingAuthManager(jwtDecoder: JwtDecoder,
```

Comment:
had to provide my own authentication manager as `JwtAuthenticationProvider` is final and can't be overridden